src/common.jl

```diff
@@ -1,194 +1,283 @@
+    integrator.kshortsize = 1
+    resize!(integrator.k, integrator.kshortsize)
+    integrator.k[1] = integrator.fsalfirst
+    # integrator.f(integrator.fsalfirst,integrator.uprev,integrator.p,integrator.t)
+    integrator.fsalfirst = constant_term.(duT)
+    integrator.destats.nf += 1
+end
+
+function perform_step!(integrator, cache::TaylorMethodCache)
+    @unpack t, dt, u, f, p = integrator
+    @unpack k, tT, uT, duT, uauxT, parse_eqs = cache
+    evaluate!(uT, dt, u)
+    tT[0] = t+dt
+    for i in eachindex(u)
+        @inbounds uT[i][0] = u[i]
+        duT[i].coeffs .= zero(duT[i][0])
     end
+    __jetcoeffs!(Val(parse_eqs.x), f, tT, uT, duT, uauxT, p)
+    k = constant_term.(duT) # For the interpolation, needs k at the updated point
+    integrator.destats.nf += 1
+end
 
-    if !isinplace && typeof(prob.u0) <: Vector{Float64}
-        f! = (du, u, p, t) -> (du .= f(u, p, t); 0)
-        if isempty(saveat_vec)
-            t, vectimeseries = taylorinteg(f!, prob.u0, prob.tspan[1], prob.tspan[2],
-                alg.order, abstol, prob.p, maxsteps=maxiters, parse_eqs=false)
-        else
-            t = saveat_vec
-            vectimeseries = taylorinteg(f!, prob.u0, t,
-                alg.order, abstol, prob.p, maxsteps=maxiters, parse_eqs=false)
-        end
+stepsize_controller!(integrator,alg::_TaylorMethod) = stepsize(integrator.cache.uT, integrator.opts.abstol)
+step_accept_controller!(integrator, alg::_TaylorMethod, q) = q
 
-    elseif !isinplace && typeof(prob.u0) <: AbstractArray
-        f! = (du, u, p, t) -> (du .= vec(f(reshape(u, sizeu), p, t)); 0)
-        if isempty(saveat_vec)
-            t, vectimeseries = taylorinteg(f!, prob.u0, prob.tspan[1], prob.tspan[2],
-                alg.order, abstol, prob.p, maxsteps=maxiters, parse_eqs=false)
-        else
-            t = saveat_vec
-            vectimeseries = taylorinteg(f!, prob.u0, t,
-                alg.order, abstol, prob.p, maxsteps=maxiters, parse_eqs=false)
-        end
+function DiffEqBase.solve(
+        prob::DiffEqBase.AbstractODEProblem{uType, tupType, isinplace},
+        alg::TaylorMethod, args...;
+        verbose=true,
+        kwargs...) where
+        {uType, tupType, isinplace}
+
+    if verbose
+        warned = !isempty(kwargs) && check_keywords(alg, kwargs, warnlist)
+        warned && warn_compat()
+    end
 
-    elseif isinplace && typeof(prob.u0) <: AbstractArray{eltype(uType),2}
-        f! = (du, u, p, t) -> (
-                dd = reshape(du, sizeu); uu = reshape(u, sizeu);
-                f(dd, uu, p, t); u = vec(uu); du = vec(dd); 0)
-        u0 = vec(prob.u0)
-        if isempty(saveat_vec)
-            t, vectimeseries = taylorinteg(f!, u0, prob.tspan[1], prob.tspan[2],
-                alg.order, abstol, prob.p, maxsteps=maxiters, parse_eqs=false)
+    sizeu = size(prob.u0)
+    # This if block handles DynamicalODEProblems
+    # credit to @SebastianM-C for coming up with this solution
+    if prob.f isa DynamicalODEFunction
```

Comment:
From what I understand, this special casing is needed because the `jetcoeffs!` function calls the user function with `xaux` which is an `Array`. I was wondering if it would be possible to just use the `DiffEqFunction` interface (i.e. call `prob.f` with `f(du,u,p,t)` and `u` with the same type as `prob.u0`), but I'm not sure I understand how the `jetcoeffs!` code works and if the above suggestion would make sense.

Reply:
Thank you for your review and your suggestions! You're right, it'd be nice if we didn't have to convert everything to an `ODEProblem`... after reading your comment here I gave it a shot, and I think just got to overcome this conversion, at least for the in-place case (I should push that commit in a couple of minutes). The out-of-place version won't work as of now, since `TaylorIntegration.jl` currently does not support oop for `prob.u0 <: AbstractArray`. But once we add that, perhaps in another PR, we'll be able to get rid of most of the `if`s and `else`s, while avoiding the conversion from `DynamicalODEProblem` to `ODEProblem`.